pkg/proxy/config/config_test.go

```diff
@@ -457,3 +460,137 @@ func TestNewEndpointsMultipleHandlersAddRemoveSetAndNotified(t *testing.T) {
 // Currently this module has a circular dependency with config, and so it's
 // named config_test, which means even test methods need to be public. This
 // is refactoring that we can avoid by resolving the dependency.
+
+type nodeTopologyHandlerMock struct {
+	topologyLabels map[string]string
+}
+
+func (n *nodeTopologyHandlerMock) OnTopologyChange(topologyLabels map[string]string) {
+	n.topologyLabels = topologyLabels
+}
+
+// waitForInvocation waits for event handler to complete processing of the invocation.
+func waitForInvocation(invoked <-chan struct{}) error {
+	for {
```

Comment:
this for loop is not really needed, the select blocks and both statement return

Reply:
ah, this got rewritten a few times in the earlier PR. the loop probably made more sense originally...